File: lib/py/uapi/internal/types/UInteger.py
from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from uapi.RandomGenerator import RandomGenerator
    from uapi.internal.validation.ValidationFailure import ValidationFailure
    from uapi.internal.types.UTypeDeclaration import UTypeDeclaration
    from uapi.internal.generation.GenerateContext import GenerateContext

from uapi.internal.types.UType import UType

_INTEGER_NAME: str = "Integer"


class UInteger(UType):

    def get_type_parameter_count(self) -> int:
        return 0

    def validate(self, value: object, select: dict[str, object] | None, fn: str | None,
                 type_parameters: list['UTypeDeclaration']) -> list['ValidationFailure']:
        from uapi.internal.validation.ValidateInteger import validate_integer
        return validate_integer(value)

    def generate_random_value(self, ctx: 'GenerateContext') -> object:
        from uapi.internal.generation.GenerateRandomInteger import generate_random_integer
        return generate_random_integer(ctx)

    def get_name(self) -> str:
        return _INTEGER_NAME

File: lib/java/src/main/java/uapi/internal/types/UInteger.java
package uapi.internal.types;

import static uapi.internal.generation.GenerateRandomInteger.generateRandomInteger;
import static uapi.internal.validation.ValidateInteger.validateInteger;

import java.util.List;

import uapi.internal.generation.GenerateContext;
import uapi.internal.validation.ValidateContext;
import uapi.internal.validation.ValidationFailure;

public class UInteger implements UType {
    public static final String _INTEGER_NAME = "Integer";

    @Override
    public int getTypeParameterCount() {
        return 0;
    }

    @Override
    public List<ValidationFailure> validate(Object value, List<UTypeDeclaration> typeParameters, ValidateContext ctx) {
        return validateInteger(value);
    }

    @Override
    public Object generateRandomValue(GenerateContext ctx) {
        return generateRandomInteger(ctx);
    }

    @Override
    public String getName() {
        return _INTEGER_NAME;
    }
}
